### ISSUE TITLE
ListLeftPushAsync doesn't use AsyncTimeout?

### ISSUE THREAD

**Issue by lastdan** [StackExchange/StackExchange.Redis#2699]
Seems like ListLeftPushAsync doesn't use an AsyncTimeout configuration option. I have async timeout set to 1000ms. But sometimes ListLeftPushAsync calls can take more than 20 seconds to finish. Shouldn't they time out earlier?

**Comment by NickCraver** [comment]
Do you have an example timeout message? We'd want to se the values here because it may be an external factor like thread pool exhaustion at those values.

**Comment by lastdan** [comment]
no, I have the next scenario) I've set AsyncTimeout to 1 second. I call ListLeftPushAsync, and sometimes this command takes 15+ seconds to execute (it finishes normally, but takes a lot of time). The expected behaviour is to see a timeout exception after a ~1 second of execution, am I right? I've alse set SyncTimeout to 1 second, and ListLeftPush sometimes throws timeout exception (as expected)

**Comment by NickCraver** [comment]
If you have an example error message, we can help. If you don't, we can't advise further. We explicitly pack so much information into those messages just to be able to answer "why?" that you're asking. Without any specifics at all, it's just guesses.

**Comment by lastdan** [comment]
ehm...what message? My question is why AsyncTimeout setting doesn't affect ListLeftPushAsync? Let's say I call ListLeftPushAsync which I know will take 15+ seconds to execute. By setting AsyncTimeout to 1 second I expect this call to end after a second with a timeout exception, but instead of this it runs for 15+ seconds. Is it by design for this method or is it a bug?
